### PR TITLE
[3.9] bpo-46416: Allow direct invocation of `Lib/test/test_typing.py` (GH-30641)

### PR DESCRIPTION
**Pull request by sobolevn**
(cherry picked from commit 2792d6d18eab3efeb71e6397f88db86e610541f1)

Co-authored-by: Nikita Sobolev <mail@sobolevn.me>

https://bugs.python.org/issue46416

### CODE REVIEW

File: Lib/test/test_typing.py [hunk 1]
```diff
@@ -4310,6 +4310,11 @@ def test_annotated_in_other_types(self):
         X = List[Annotated[T, 5]]
         self.assertEqual(X[int], List[Annotated[int, 5]])
 
+    def test_annotated_mro(self):
```
Comment (JelleZijlstra) [inline Lib/test/test_typing.py:4313]: This looks unrelated to your PR title, any idea what happened?

Comment (sobolevn) [reply Lib/test/test_typing.py:4313]: Looks like I backported a different commit 🤔 

Thanks for noticing!

Comment (sobolevn) [reply Lib/test/test_typing.py:4313]: I got the same result second time: https://github.com/python/cpython/pull/30707
